File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        
        count = 0
        left = right = 0
        
        while right < len(nums) - 1:
            
            furthest = 0
            
            for i in range(left , right + 1):
                
                furthest = max(furthest, i + nums[i] )
                
            left = right + 1
            right = furthest
            count += 1
            
        return count

File: 0045-jump-game-ii/0045-jump-game-ii.go
import "math"

func jump(nums []int) int {
	count := 0
	left, right := 0, 0

	for right < len(nums)-1 {
		furthest := 0

		for i := left; i <= right; i++ {
			furthest = int(math.Max(float64(furthest), float64(i+nums[i])))
		}

		left = right + 1
		right = furthest
		count++
	}

	return count
}